src/datumaro/experimental/fields/masks.py

```diff
@@ -32,6 +33,14 @@ class MaskField(Field):
     channels_first: bool = False
     has_channels_dim: bool = False
 
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        if not (self.dtype.is_(pl.Boolean) or self.dtype.is_unsigned_integer()):
+            raise ValueError(
+                "A mask field's dtype has to be a Boolean or an unsigned integer dtype (e.g. UInt8). These integers "
+                "refer to mask color info provided in mask categories that is defined in a dataset's schema."
+            )
```

Comment:
```suggestion
            raise ValueError("MaskField's dtype must be a polars Boolean or unsigned integer type (e.g. UInt8).")
```
Since the definition of a Field/Sample does not depend on Dataset, I would omit the second sentence because it could generate confusion - the user can build a wrong-typed sample and trigger this exception even without creating a Dataset.

Reply:
Given the purpose of datumaro, I think it is safe to assume users will eventually create a dataset, and given that we both fix the dtype and require categories, I think we should be clear about the choices that we made and the intended use of a field/category/dataset. 

As a user, I would first assume a label field would be a string dtype, so knowing where to put what information would be useful to me. This also comes back to our previous discussion, where I suggested to add the categories to the Sample class and use that as a schema. Now to some extend the sample does not have meaning until it is added to a dataset, and depending on the schema, the same sample could have different meaning in different datasets. 

Your suggested error message only helps users who already understand the meaning, and put for example Int32, but won't help users who try to create a field with string or object dtype.

Reply:
I see where you're coming from, the string scenario is a good example of potential common mistake which I didn't consider. I suggest this description, what do you think?
`"A mask field's dtype must be a polars Boolean or unsigned integer type (e.g. UInt8). This integer normally represents the index of a category, with reference to the mask categories of the the dataset to which the sample is (or will be) appended."`